azurerm/resource_arm_iot_dps_cerificate.go

```diff
@@ -0,0 +1,149 @@
+				ForceNew:     true,
+				ValidateFunc: validate.IoTHubName,
+			},
+
+			"certificate_content": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 10000),
+			},
+		},
+	}
+}
+
+func resourceArmIotDPSCertificateCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).iothub.DPSCertificateClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	iotDPSName := d.Get("iot_dps_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, name, resourceGroup, iotDPSName, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_iot_dps_certificate", *existing.ID)
+		}
+	}
+
+	certificate := iothub.CertificateBodyDescription{
+		Certificate: utils.String(d.Get("certificate_content").(string)),
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, iotDPSName, name, certificate, ""); err != nil {
+		return fmt.Errorf("Error creating/updating IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, name, resourceGroup, iotDPSName, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmIotDPSCertificateRead(d, meta)
+}
+
+func resourceArmIotDPSCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).iothub.DPSCertificateClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	iotDPSName := id.Path["provisioningServices"]
+	name := id.Path["certificates"]
+
+	resp, err := client.Get(ctx, name, resourceGroup, iotDPSName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving IoT Device Provisioning Service Certificate %q (Device Provisioning Service %q / Resource Group %q): %+v", name, iotDPSName, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("iot_dps_name", iotDPSName)
+
```

Comment:
can we add a comment here to clarify that `certificate_content` isn't returned from the API?